presto-hive/src/main/java/com/facebook/presto/hive/parquet/HdfsParquetDataSource.java

```diff
@@ -72,16 +79,20 @@ public final void readFully(long position, byte[] buffer)
     @Override
     public final void readFully(long position, byte[] buffer, int bufferOffset, int bufferLength)
     {
-        readInternal(position, buffer, bufferOffset, bufferLength);
         readBytes += bufferLength;
+        long readStart = System.nanoTime();
+
+        readInternal(position, buffer, bufferOffset, bufferLength);
+
+        long currentReadTimeNanos = System.nanoTime() - readStart;
+        readTimeNanos += currentReadTimeNanos;
```

Comment:
I'd just use `Stopwatch` it can be started and stopped repeatedly.

Reply:
@martint  what about:
```
Stopwatch stopwatch = Stopwatch.createStarted();
readInternal(...);
long elapsed = stopwatch.elapsed(NANOSECONDS);

readTimeNanos += elapsed;
...
```
?

I don't think this is a place where allocating an additional object is a problem. I don't have a strong opinion though.

Reply:
Seems both agree to use simple calculation without stopwatch. I will change that back then.